### PR TITLE
Add pystackreg recipe

### PR DESCRIPTION
**Pull request by mrakitin**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by mrakitin** [comment]
@glichtner, do you wish to co-maintain this feedstock? Should I add Philippe Thevenaz?

I have a few questions:
1) What's the canonical repository: https://github.com/glichtner/pystackreg or https://bitbucket.org/glichtner/pystackreg?
2) What's the type of the license for the project? It looks like it's "Custom", but I am not fully sure. Maybe @conda-forge/core could figure out what to put to the recipe?

Thank you.

File: recipes/pystackreg/LICENSE.txt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+|
+| Other relevant on-line publications are available at
+| http://bigwww.epfl.ch/publications/
+\===================================================================*/
+
+/*====================================================================
+| Additional help available at http://bigwww.epfl.ch/thevenaz/turboreg/
+|
+| You'll be free to use this software for research purposes, but you
+| should not redistribute it without our consent. In addition, we expect
```
Comment (jakirkham) [inline recipes/pystackreg/LICENSE.txt]: Based on this line, it sounds like we cannot distribute it. Do they consent to placing this on conda-forge? That could mean it is used for things outside of "research purposes". Is that ok with them? Also later lines go on to request a citation or acknowledgement in published results. What if it is used for something that doesn't have results? Or what if it is used in an application that produces results, but whose results cannot be shared publicly?

For me personally I find this sort of license hard to wrap my head around as it is not a standard OSS license. It seems like we could wind up in trouble or are users could. Though I might be overly worried about this.

Perhaps others can comment on this.

cc @conda-forge/core @conda-forge/staged-recipes

File: recipes/pystackreg/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+  host:
+    - python
+    - numpy
+    - pip
+  run:
+    - python
+    - numpy
```
Comment (chrisburr) [inline recipes/pystackreg/meta.yaml]: I presume this is linking against numpy. If so, see https://conda-forge.org/docs/maintainer/knowledge_base.html#building-against-numpy

File: recipes/pystackreg/LICENSE.txt (hunk 1 above)
Comment (chrisburr) [reply recipes/pystackreg/LICENSE.txt]: I also think it forbids conda-forge from distributing this.

@mrakitin Can you contact the authors to ask them to endorse coda-forge? Or better yet, use a standard OSS licence

**Comment by glichtner** [comment]
> 
> 
> @glichtner, do you wish to co-maintain this feedstock? Should I add Philippe Thevenaz?
> 
> I have a few questions:
> 
>     1. What's the canonical repository: https://github.com/glichtner/pystackreg or https://bitbucket.org/glichtner/pystackreg?
> 
>     2. What's the type of the license for the project? It looks like it's "Custom", but I am not fully sure. Maybe @conda-forge/core could figure out what to put to the recipe?
> 
> 
> Thank you.

Thanks for inlcuding me. I'm happy to co-maintain this.
Regarding your questions:
1. The canonical repository is now https://github.com/glichtner/pystackreg
2. As this software is only a port of the original Fiji package, I included the license of that package after asking Mr Thévenaz for permission (that was my only contat with him so far) - his reply was to "If you distribute your framework publicly, make sure a reference to one of our papers is on display somewhere". I'll ask him whether he agrees to put the package on conda-forge if the documentation of the package itself contains a reference to his paper (but not necessarily the software that is created with the package).

File: recipes/pystackreg/LICENSE.txt (hunk 1 above)
Comment (glichtner) [reply recipes/pystackreg/LICENSE.txt]: I am the author of this python package, which itself is just a port of the "stackreg" package written in Java by Philippe Thévenaz. I will ask the author of the original package if he endorses conda-forge and get back to you.

File: recipes/pystackreg/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+test:
+  imports:
+    - pystackreg
+
+about:
+  home: https://bitbucket.org/glichtner/pystackreg
+  license: Custom
+  license_family: OTHER
+  license_file: LICENSE.txt
+  summary: Python implementation of the ImageJ/FIJI Plugin TurboReg/StackReg
```
Comment (chrisburr) [inline recipes/pystackreg/meta.yaml]: You can include the citation in the README and Anaconda.org page using something like
```suggestion
  summary: Python implementation of the ImageJ/FIJI Plugin TurboReg/StackReg
  description: |
    Some further description...
 
    Based in the work in X.
```

File: recipes/pystackreg/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    - python
+    - numpy
+    - tqdm
+
+test:
+  imports:
+    - pystackreg
+
+about:
+  home: https://bitbucket.org/glichtner/pystackreg
```
Comment (chrisburr) [inline recipes/pystackreg/meta.yaml]:
```suggestion
  home: https://github.com/glichtner/pystackreg
```

**Comment by mrakitin** [comment]
Thank you, @glichtner, I added you as a maintainer. Looking forward to the reply from the author of the original software.

File: recipes/pystackreg/meta.yaml (hunk 2 above)
Comment (mrakitin) [reply recipes/pystackreg/meta.yaml]: Done, thanks for the hint!

**Comment by glichtner** [comment]
Mr Thévenaz agreed to change the license in a way that we hope enables the package to be distributed via conda-forge. I have updated the package's license on github, it is now:

> You are free to use this software for commercial and non-commercial purposes. However, we expect you to include a citation or acknowledgement whenever you present or publish research results that are based on this software. You are free to modify this software or derive works from it, but you are only allowed to distribute it under the same terms as this license specifies. Additionally, you must include a reference to the research paper above in all software and works derived from this software.

**Comment by CJ-Wright** [comment]
This sounds like a citation version of the GPL. What does "works derived from this software" mean?

File: recipes/pystackreg/meta.yaml (hunk 3 above)
Comment (mrakitin) [reply recipes/pystackreg/meta.yaml]: Added. See below.

**Comment by mrakitin** [comment]
@glichtner, thanks for clarifying the license question and publishing a new version 0.2.2 on PyPI. I updated the recipe to use the latest version as well as added the description with the citation info. Does it look fine to you?

I also noticed that the PyPI, https://github.com/glichtner/pystackreg/pull/4 is going to address it.

@conda-forge/staged-recipes, do the license and its type look OK to you now?

**Comment by glichtner** [comment]
@mrakitin Thanks for your efforts, looks fine to me!

**Comment by mrakitin** [comment]
@conda-forge/staged-recipes, are there any other concerns preventing it from merging? Thanks.

**Comment by mrakitin** [comment]
@conda-forge/staged-recipes, I don't want to lose this package. Could you please merge while it's still alive?

**Comment by chrisburr** [comment]
I think the license looks fine now but I'll give others a chance to comment before merging.

As this PR is old I'm going to rerun the CI to make sure it still passes: @conda-forge-admin, please restart ci

(do feel free to ping us more often, once a few days have passed its unlikely anyone will notice without being prompted)

File: recipes/pystackreg/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    - tqdm
+
+test:
+  imports:
+    - pystackreg
+
+about:
+  home: https://github.com/glichtner/pystackreg
+  license: Custom
+  license_family: OTHER
```
Comment (jakirkham) [inline recipes/pystackreg/meta.yaml:37]: Maybe we can do something like this? `license_family` would be filled in with repodata patching so is optional.

```suggestion
  license: LicenseRef-TurboReg-StackReg
```

**Comment by jakirkham** [comment]
Linux appears to be failing for some reason. Tried to restart it, but it is still failing.

**Comment by mrakitin** [comment]
That's weird, I haven't observed it before. Something new, maybe introduced by the strict channel priority?

File: recipes/pystackreg/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+  host:
+    - python
+    - numpy 1.16
```
Comment (chrisburr) [inline recipes/pystackreg/meta.yaml:23]: In the feedstock this should be changed back to
```suggestion
    - numpy
```
and then be rerendered.

Comment (mrakitin) [reply recipes/pystackreg/meta.yaml]: Oh, ups. Did not realize it should not go here. Will fix it.

**Comment by mrakitin** [comment]
Thank you, @chrisburr!

**Pull request merged**